lib/services/unified_resource_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package services_test
+
+import (
+	// "context"
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/keystore"
+	"github.com/gravitational/teleport/lib/auth/testauthority"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	"github.com/gravitational/teleport/lib/observability/tracing"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local"
+)
+
+func setupConfig(t *testing.T, ctx context.Context) (auth.InitConfig, *memory.Memory) {
+	tempDir := t.TempDir()
+
+	clock := clockwork.NewFakeClock()
+
+	bk, err := memory.New(memory.Config{Context: ctx, Clock: clock})
+	require.NoError(t, err)
+
+	clusterName, err := services.NewClusterNameWithRandomID(types.ClusterNameSpecV2{
+		ClusterName: "me.localhost",
+	})
+	require.NoError(t, err)
+
+	return auth.InitConfig{
+		DataDir:                 tempDir,
+		HostUUID:                "00000000-0000-0000-0000-000000000000",
+		NodeName:                "foo",
+		Backend:                 bk,
+		Authority:               testauthority.New(),
+		ClusterAuditConfig:      types.DefaultClusterAuditConfig(),
+		ClusterNetworkingConfig: types.DefaultClusterNetworkingConfig(),
+		SessionRecordingConfig:  types.DefaultSessionRecordingConfig(),
+		ClusterName:             clusterName,
+		StaticTokens:            types.DefaultStaticTokens(),
+		AuthPreference:          types.DefaultAuthPreference(),
+		SkipPeriodicOperations:  true,
+		KeyStoreConfig: keystore.Config{
+			Software: keystore.SoftwareConfig{
+				RSAKeyPairSource: testauthority.New().GenerateKeyPair,
+			},
+		},
+		Tracer: tracing.NoopTracer(teleport.ComponentAuth),
+	}, bk
+}
+func TestUnifiedResourceWatcher(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	conf, bk := setupConfig(t, ctx)
+	authServer, err := auth.Init(ctx, conf)
```

Comment:
I need the authserver as the `ResourceGetter` because the `presence` doesn't have access to all the getters I need (like windowsdesktopgetter). Am I misunderstanding the intent? (the test was updated a bit since this comment)